src/btree/bt_vrfy_dsk.c

```diff
@@ -237,6 +237,89 @@ __verify_dsk_value_validity(WT_SESSION_IMPL *session, WT_CELL_UNPACK_KV *unpack,
+     * aggregate. First check that the aggregate is consistent with being deleted at the given time.
+     */
+    if (unpack->ta.newest_stop_durable_ts > unpack->page_del.durable_timestamp)
+        WT_RET_VRFY(session,
+          "fast-delete cell %" PRIu32
+          " on page at %s has invalid newest durable stop time; should be <= %" PRIu64
+          "; time aggregate %s",
+          cell_num - 1, tag, unpack->page_del.durable_timestamp,
+          __wt_time_aggregate_to_string(&unpack->ta, time_string));
+    if (unpack->ta.newest_txn > unpack->page_del.txnid)
+        WT_RET_VRFY(session,
+          "fast-delete cell %" PRIu32
+          " on page at %s has invalid newest transaction; should be <= %" PRIu64
+          "; time aggregate %s",
+          cell_num - 1, tag, unpack->page_del.txnid,
+          __wt_time_aggregate_to_string(&unpack->ta, time_string));
+    if (unpack->ta.newest_stop_ts != WT_TS_MAX &&
+      unpack->ta.newest_stop_ts > unpack->page_del.timestamp)
+        WT_RET_VRFY(session,
+          "fast-delete cell %" PRIu32
+          " on page at %s has invalid newest stop time; should be <= %" PRIu64
+          "; time aggregate %s",
+          cell_num - 1, tag, unpack->page_del.timestamp,
+          __wt_time_aggregate_to_string(&unpack->ta, time_string));
+    if (unpack->ta.newest_stop_txn != WT_TXN_MAX &&
+      unpack->ta.newest_stop_txn > unpack->page_del.txnid)
+        WT_RET_VRFY(session,
+          "fast-delete cell %" PRIu32
+          " on page at %s has invalid newest stop transaction; should be <= %" PRIu64
+          "; time aggregate %s",
+          cell_num - 1, tag, unpack->page_del.txnid,
+          __wt_time_aggregate_to_string(&unpack->ta, time_string));
+
+    /*
+     * Merge this information into the aggregate and verify the results, against the parent if
+     * possible.
+     */
+    WT_TIME_AGGREGATE_COPY(&ta_with_delete, &unpack->ta);
+    ta_with_delete.newest_stop_durable_ts = unpack->page_del.durable_timestamp;
+    ta_with_delete.newest_txn = unpack->page_del.txnid;
+    ta_with_delete.newest_stop_ts = unpack->page_del.timestamp;
+    ta_with_delete.newest_stop_txn = unpack->page_del.txnid;
+    ret = __wt_time_aggregate_validate(session, &ta_with_delete, addr != NULL ? &addr->ta : NULL,
+      F_ISSET(session, WT_SESSION_QUIET_CORRUPT_FILE));
+    if (ret != 0)
+        WT_RET_VRFY_RETVAL(session, ret,
+          "fast-delete cell %" PRIu32 " on page at %s failed adjusted timestamp validation",
+          cell_num - 1, tag);
+
+    /*
+     * The other elements of the structure are not stored on disk and are set unconditionally by the
+     * unpack code, so just assert about them. Prepared fast-truncates are not allowed to be
+     * evicted.
+     */
+    WT_ASSERT(session, unpack->page_del.prepare_state == 0);
+    WT_ASSERT(session, unpack->page_del.previous_ref_state == WT_REF_DISK);
+    WT_ASSERT(session, unpack->page_del.committed == true);
```

Comment:
Verify is also supposed to detect corruption, so these shouldn't be asserts, they should be "normal" failures.

Reply:
As per the comment, those aren't read from disk, so asserts seemed like the right thing...

Reply:
Got it, makes sense.